tools/hsmtool.c

```diff
@@ -503,7 +503,9 @@ static int generate_hsm(const char *hsm_secret_path)
 	if (fd < 0) {
 		errx(ERROR_USAGE, "Unable to create hsm_secret file");
 	}
-	if (!write_all(fd, bip32_seed, bip32_seed_len))
+	/* Write only the first 32 bytes, length of the (plaintext) seed in the
+	 * hsm_secret. */
+	if (!write_all(fd, bip32_seed, 32))
```

Comment:
Out of curiosity, what are the remaining 32 bytes returned by `bip39_mnemonic_to_seed`?

Reply:
It's the output of a SHA512 https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#from-mnemonic-to-seed .
I'm still confused about how i could overlook this in my initial review...

Reply:
Thinking more about it i'm not sure it's cryptographically sound. Would `sha256`ing the output be preferable ? This would completely break the link with previously-generated seeds, but hey they were not even valid in the first place.

Reply:
Hm, switching hash function seems the wrong way to go here. Let's stay as close to bip39 as possible. Can we even still express our derivation paths as BIP32 HD keys? Would be quite inconvenient if we break compatibility with all existing recovery tools.

Reply:
We don't have compatibility with existing recovery tools already, as we use bip39 to derive the seed which we'll later derive the actual BIP32 seed from (with our homebrewed HKDF).